impexp-core/src/main/java/org/citydb/database/connection/DatabaseConnectionPool.java

```diff
@@ -192,22 +221,32 @@ public AbstractDatabaseAdapter getActiveDatabaseAdapter() {
 	}
 
 	public Connection getConnection() throws SQLException {
-		if (!isConnected())
+		if (!isConnected()) {
 			throw new SQLException("Database is not connected.");
+		}
 
 		Connection connection = dataSource.getConnection();
 		connection.setAutoCommit(true);
 
+		// change workspace if required
+		if (databaseAdapter.hasVersioningSupport()) {
+			Workspace workspace = databaseAdapter.getConnectionDetails().getWorkspace();
+			databaseAdapter.getWorkspaceManager().gotoWorkspace(connection, workspace);
```

Comment:
I added `databaseAdapter.getConnectionDetails().isSetWorkspace()`  to avoid a NPE (see 28e509be11dd54ce92c453eafba050843e6c5149).

Yes, I think it's necessary to call `gotoWorkspace` every time. At least, this is the behaviour in `master` and previous releases. As far as I remember the reason is the use of a connection pool. Assume you connect to the database with workspace `ABC` and execute some operations. Afterwards, you disconnect from the database. Due to the connection pool, the connection is not physically closed. If you now re-connect, for instance, with the default workspace, the connection pool might return the previous connection from `getConnection` that is still on workspace `ABC`. So, you must invoke `gotoWorkspace` to set it to the new workspace correctly.